packages/core/strapi/tests/api/populate/filtering/index.test.api.js

```diff
@@ -315,4 +315,64 @@ describe('Populate filters', () => {
       expect(body.data[0].attributes.third).toBeUndefined();
     });
   });
+
+  describe('Populate a dynamic zone', () => {
+    test('Populate every component in the dynamic zone', async () => {
+      const qs = {
+        populate: {
+          dz: '*',
+        },
+      };
+
+      const { status, body } = await rq.get(`/${schemas.contentTypes.b.pluralName}`, { qs });
+
+      expect(status).toBe(200);
+      expect(body.data).toHaveLength(2);
+      expect(body.data[0].attributes.dz).toHaveLength(3);
+      expect(body.data[1].attributes.dz).toHaveLength(1);
+    });
+
+    test('Populate only one component type using fragment', async () => {
+      const qs = {
+        populate: {
+          dz: {
+            on: {
+              'default.foo': true,
```

Comment:
should this be true or * here ?

Reply:
Since we don't have attributes that can be populated in the components, I think that true makes more sense, although '*' should work too

Reply:
I was wondering if true even worked but it was recently added If I remember correctly. You can just add some check in the test to verify the data is valid not just that there is one item maybe